master/internal/user/postgres_users.go

```diff
@@ -101,18 +142,200 @@ func AddUserExec(user *model.User) error {
 		GroupID: personalGroup.ID,
 	}
 	if _, err = tx.NewInsert().Model(&groupMembership).Exec(ctx); err != nil {
-		return errors.Wrap(err, "error adding user to personal group")
+		return 0, errors.Wrap(err, "error adding user to personal group")
 	}
 
 	if err = tx.Commit(); err != nil {
-		return errors.Wrap(err, "error committing changes in AddUserExec")
+		return 0, errors.Wrap(err, "error committing changes in addUserExec")
+	}
+
+	return user.ID, nil
+}
+
+// AddUser creates a new user.
+func AddUser(
+	ctx context.Context,
+	user *model.User,
+	ug *model.AgentUserGroup,
+) (model.UserID, error) {
+	userID, err := addUserExec(ctx, user)
+	if err != nil {
+		return 0, err
+	}
+
+	if ug != nil {
+		if err := addAgentUserGroup(ctx, userID, ug); err != nil {
+			return 0, err
+		}
+	}
+	return userID, nil
+}
+
+// UpdateUser updates an existing user.  `toUpdate` names the fields to update.
+func UpdateUser(
+	ctx context.Context,
+	updated *model.User,
+	toUpdate []string,
+	ug *model.AgentUserGroup,
+) error {
+	tx, err := db.Bun().BeginTx(ctx, nil)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		txErr := tx.Rollback()
+		if txErr != nil && txErr != sql.ErrTxDone {
+			log.WithError(txErr).Error("error rolling back transaction in addUserExec")
+		}
+	}()
+
+	if len(toUpdate) > 0 {
+		_, err := tx.NewUpdate().Model(updated).Column(
+			strings.Join(toUpdate, ",")).Where("id = ?", updated.ID).Exec(ctx)
+		if err != nil {
+			return errors.Wrapf(err, "updating %q", updated.Username)
+		}
 	}
 
+	var updatePassword bool
+	for _, e := range toUpdate {
+		if e == "password_hash" {
+			updatePassword = true
+			break
+		}
+	}
+
+	if updatePassword {
+		_, err = tx.NewDelete().Model(updated).Where("user_id = ?", updated.ID).Exec(ctx)
+		if err != nil {
+			return errors.Wrap(err, "error deleting user sessions")
+		}
+	}
+
+	if ug != nil {
+		if err = deleteAgentUserGroup(ctx, &tx, updated.ID); err != nil {
+			return err
+		}
+		if *ug != (model.AgentUserGroup{}) {
+			if err = addAgentUserGroup(context.TODO(), updated.ID, ug); err != nil {
+				return err
+			}
+		}
+	}
+
+	if err = tx.Commit(); err != nil {
+		return errors.Wrap(err, "error committing changes to user in UpdateUser")
+	}
+
+	return nil
+}
+
+// UserList returns all of the users in the database.
+func UserList(ctx context.Context) (values []model.FullUser, err error) {
+	query, err := os.ReadFile("../../static/srv/list_users.sql")
+	if err != nil {
+		return nil, err
+	}
+
+	err = db.Bun().NewRaw(string(query)).Scan(ctx, &values)
+	return values, err
+}
+
+// UserImage returns the profile picture associated with the user.
+func UserImage(ctx context.Context, username string) (photo []byte, err error) {
+	type photoRow struct {
+		Photo []byte
+	}
+	var userPhoto photoRow
+	query, err := os.ReadFile("../../static/srv/get_user_image.sql")
+	if err != nil {
+		return nil, err
+	}
+	err = db.Bun().NewRaw(string(query), username).Scan(ctx, &userPhoto)
+	return userPhoto.Photo, err
```

Comment:
I wasn't sure, since the queries here are written in a different file it's less transparent if there are any changes to reflect that in a specific Bun query. Instead, I used `NewRaw` as my catch-all

Reply:
Yeah, the queries are in a different file just because that is how we wrote queries for a while (see all the QueryProtof infra for more history). Now that we have consolidated around bun, we want them to just be bun.

Reply:
so should I delete the .sql files too? Otherwise i think it would be confusing to the user to see a bun query in one file but then have the opportunity to change an unused but similar-looking 'query' in a sql file elsewhere

Reply:
yep, if you bunify this, delete them.